include/secp256k1_schnorr_adaptor.h

```diff
@@ -0,0 +1,140 @@
+#ifndef SECP256K1_SCHNORR_ADAPTOR_H
+#define SECP256K1_SCHNORR_ADAPTOR_H
+
+#include "secp256k1.h"
+#include "secp256k1_extrakeys.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** A pointer to a function to deterministically generate a nonce.
+ *
+ *  Same as secp256k1_schnorrsig_nonce function with the exception of accepting an
+ *  additional adaptor point t argument. The adaptor point argument can protect
+ *  signature schemes with key-prefixed challenge hash inputs against reusing the
+ *  nonce when signing with different adaptor points.
+ *
+ *  Returns: 1 if a nonce was successfully generated. 0 will cause signing to
+ *           return an error.
+ *  Out:  nonce32: pointer to a 32-byte array to be filled by the function
+ *  In:     msg32: the 32-byte message being verified (will not be NULL)
+ *          key32: pointer to a 32-byte secret key (will not be NULL)
+ *      xonly_t32: the 32-byte serialized xonly adaptor point (will not be NULL)
+ *     xonly_pk32: the 32-byte serialized xonly pubkey corresponding to key32
+ *                 (will not be NULL)
+ *           algo: pointer to an array describing the signature
+ *                 algorithm (will not be NULL)
+ *        algolen: the length of the algo array
+ *           data: arbitrary data pointer that is passed through
+ *
+ *  Except for test cases, this function should compute some cryptographic hash of
+ *  the message, the key, the adaptor point, the pubkey, the algorithm description, and data.
+ */
+typedef int (*secp256k1_adaptor_nonce_function_hardened)(
+    unsigned char *nonce32,
+    const unsigned char *msg32,
+    const unsigned char *key32,
+    const unsigned char *xonly_t32,
```

Comment:
```suggestion
    const unsigned char *t33,
```
It is better to pass the compressed adaptor pubkey for nonce creation (without ignoring the parity byte)

Reply:
Changed in the new commit